backend/de.metas.manufacturing/src/main/java/org/eevolution/productioncandidate/service/PPOrderCandidateService.java

```diff
@@ -368,4 +377,47 @@ private void syncOrderLineWithBOMLine(
 
 		ppOrderCandidateDAO.saveLine(orderCandidateLine);
 	}
+
+	public void recomputeMaturingCandidates()
+	{
+		ppMaturingCandidatesViewRepo.deleteStaleCandidates();
+		ppMaturingCandidatesViewRepo.streamValidCandidates()
+				.forEach(this::createUpdateCandidate);
+	}
+
+	private void createUpdateCandidate(final @NonNull PPMaturingCandidateV ppMaturingCandidatesV)
+	{
+		final ProductPlanningId productPlanningId = ppMaturingCandidatesV.getProductPlanningId();
+
+		final ProductPlanning productPlanning = productPlanningDAO.getById(productPlanningId);
+		if (!productPlanning.isMatured())
+		{
+			logger.warn("PP_Product_Planning_ID: {} is not matured", productPlanningId);
+			return;
+		}
+		if (productPlanning.getPlantId() == null)
+		{
+			logger.warn("No S_Resource_ID defined for PP_Product_Planning_ID:{}", productPlanningId);
```

Comment:
consider adding this check to ProductPlanning constructor. i.e. if matured than Plant shall be set.
Also make sure there is a model interceptor on PP_Product_Planning which tries to convert the I_PP_Product_Planning record to ProductPlanning (for validation purposes).

Reply:
added. MI validation already happens `de.metas.inoutcandidate.modelvalidator.PP_Product_Planning#beforeSave`